File: algorithms/sorting/bubble/bubble.py
def bubble(arr):
  while True:
    swapped=False
    i=0
    while i+1<len(arr):
      if arr[i]>arr[i+1]:
        arr[i],arr[i+1]=arr[i+1],arr[i]
        swapped=True
      i+=1
    if not swapped:
      return arr

print(bubble([1,4,5,2,3,6]))

File: algorithms/sorting/bubble/bubble.cpp
#include <stdio.h>

void bubble(int* arr, int size) {
  while (true) {
    bool swapped = false;
    int i = 0;

    while (i+1<size) {
      if (arr[i]>arr[i+1]) {
        int temp = arr[i];
        arr[i] = arr[i+1];
        arr[i+1] = temp;
        swapped=true;
      }
      i++;
    }

    if (!swapped) {
      return;
    }
  }
}

int main() {
  int num[] = {7, 4, 2, 5, 6};
  bubble(num, 5);
  
  for (int i = 0; i < 5; i++) {
    printf("%d ", num[i]);
  }
  printf("\n");
}